pallets/utxo/src/rewards.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): C. Yap
+
+use crate::{Config, Pallet, Event, BlockAuthorRewardAmount, BlockAuthor, MLTCoinsAvailable, TransactionOutput, UtxoStore, StartingPeriod};
+
+use frame_support::traits::Get;
+use sp_runtime::traits::{SaturatedConversion, Saturating, BlakeTwo256, Hash, Zero};
+
+
+/// Returns `true` if reward reduction period has passed; and if a new period has to be created.
+pub(crate) fn period_elapsed<T:Config>(time_now: T::BlockNumber) -> bool {
+    let starting_period =  <StartingPeriod<T>>::get();
+    let time_elapsed = time_now.saturating_sub(starting_period);
+    let has_elapse = time_elapsed > T::RewardReductionPeriod::get();
+
+    if has_elapse {
+        log::debug!("period has elapsed. Updating the start of period to now");
+        <StartingPeriod<T>>::put(time_now);
+    }
+    has_elapse
+}
+
+/// Returns the newly reduced reward amount for a Block Author.
+/// How much a reward is reduced, is based on the config's`RewardReductionFraction`.
+pub(super) fn update_reward_amount<T:Config>(coins_available:u128) -> u128 {
```

Comment:
Does this return the value in MLT or in base units?

Reply:
it will be based on the GenesisConfig of the Utxo,  set in the `chain_spec.rs`  , the `initial_reward_amount`, which is `100 * MLT_UNIT`. 

I'm assuming the rewards is always at MLT, or `1_000 * 100_000_000`